### PR TITLE
Documentation: Easier language switching

### PR DESCRIPTION
**Pull request by HypnosNova**
When readers read document and switch languages, we should update the docs he read into the language he chooses. And when he inputs some filters and change language, we should also show the same results but in different language.

### CODE REVIEW

**Comment by Mugen87** [comment]
For testing: https://raw.githack.com/HypnosNova/three.js/docs/docs/index.html

File: docs/index.html [hunk 1]
```diff
@@ -69,6 +69,8 @@ <h1><a href="http://threejs.org">three.js</a> / docs</h1>
 
 				language = value;
 				createNavigation();
+				updateFilter();
```
Comment (Mugen87) [inline docs/index.html]: This line does not seem to have an effect. When changing the language, no filtering is applied and you see all side navigation menu items.

File: docs/index.html [hunk 2]
```diff
@@ -69,6 +69,8 @@ <h1><a href="http://threejs.org">three.js</a> / docs</h1>
...
+				autoChangeUrlLanguage(language);
```
Comment (Mugen87) [inline docs/index.html]: Code style:
```js
autoChangeUrlLanguage( language );
```

**Review by Mugen87 — CHANGES_REQUESTED** [review]

File: docs/index.html (hunk 1 above)
Comment (HypnosNova) [reply docs/index.html]: when users input something in filter, for example they input geometry. So the menu items shows only about geometry. At that time when they change language, if we don't use updateFilter();, the menu list shows all menu items.

**Review by Mugen87 — APPROVED** [review]

**Comment by Mugen87** [comment]
Nice change. Much better than before 👍

File: docs/index.html [hunk 3]
```diff
@@ -226,6 +228,23 @@ <h1><a href="http://threejs.org">three.js</a> / docs</h1>
 
 			}
 
+			// Auto change language url. If a reader open a document in English, when he click "zh", the document he read will auto change into Chinese version
+
+			function autoChangeUrlLanguage( language ) {
+
+				var hash = location.hash;
+				if( !hash ){
```
Comment (mrdoob) [inline docs/index.html]:
```js
if ( hash === '' ) return;
```

**Review by mrdoob — CHANGES_REQUESTED** [review]

File: docs/index.html (hunk 3 above)
Comment (HypnosNova) [reply docs/index.html]: fixed

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**